hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientMapReduceProxy.java

```diff
@@ -269,25 +261,10 @@ public boolean cancel(boolean mayInterruptIfRunning) {
         }
 
         @Override
-        public boolean isCancelled() {
-            return cancelled;
-        }
-
-        @Override
-        public void setResult(Object result) {
+        protected void setResult(Object result) {
             super.setResult(result);
-            latch.countDown();
```

Comment:
It's kind of an SCJP-style problem :) 
These are two independent inner classes (ClientJob & ClientCompletableFuture) that are based in one outer class (ClientMapReduceProxy).
ClientJob can't access a protected method from ClientCompletableFuture's super class (AbstractCompletableFuture) since it's neither in AbstractCompletableFuture's package nor in it's inheritance tree.
However, ClientJob could access ClientCompletableFuture's protected method since they are both located in the same outer class, thus we need this setResult() method.